test/e2e/l7networkpolicy_test.go

```diff
@@ -429,17 +435,35 @@ type L7LogAlertEntry struct {
 }
 
 type L7LogEntry struct {
-	EventType   string           `json:"event_type"`
-	DestIP      string           `json:"dest_ip"`
-	DestPort    int32            `json:"dest_port"`
-	Protocol    string           `json:"proto"`
-	AppProtocol string           `json:"app_proto,omitempty"`
-	Http        *L7LogHttpEntry  `json:"http,omitempty"`
-	Alert       *L7LogAlertEntry `json:"alert,omitempty"`
+	EventType           string           `json:"event_type"`
+	DestIP              string           `json:"dest_ip"`
+	DestPort            int32            `json:"dest_port"`
+	Protocol            string           `json:"proto"`
+	AppProtocol         string           `json:"app_proto,omitempty"`
+	PacketBytes         []byte           `json:"packet,omitempty"`
+	Http                *L7LogHttpEntry  `json:"http,omitempty"`
+	Alert               *L7LogAlertEntry `json:"alert,omitempty"`
+	expectedPacketRegex *regexp.Regexp
 }
 
-func (e *L7LogEntry) Equal(x *L7LogEntry) bool {
-	return reflect.DeepEqual(e, x)
+// Matches the two L7LogEntry. They must be exactly equal if none are expected
+// log entry. Otherwise, Packet field is matched separately. The actual log entry
+// with PacketBytes must contain packet info from the other expected log entry.
+func (e *L7LogEntry) Matches(x *L7LogEntry) bool {
+	packetMatch := func(e, x *L7LogEntry) bool {
+		if e.expectedPacketRegex != nil {
+			return e.expectedPacketRegex.MatchString(string(x.PacketBytes))
+		}
+		if x.expectedPacketRegex != nil {
+			return x.expectedPacketRegex.MatchString(string(e.PacketBytes))
+		}
+		// Compare packet bytes if neither e nor x are expected log entry.
+		return bytes.Equal(e.PacketBytes, x.PacketBytes)
```

Comment:
I don't think that's the logic I described in my last review.

Reply:
Oh I misunderstood, didn't consider there to be two expected log entries, but yeah it would be more comprehensive.

Reply:
Oh wait, is the only difference between the logics "check if both `L7LogEntry` contain `expectedPacketRegex`"? Then in that case, we should not compare any `PacketBytes`? (Am I missing something)